cdap-watchdog/src/main/java/co/cask/cdap/logging/save/KafkaMessageCallback.java

```diff
@@ -66,12 +71,17 @@ public void onReceived(Iterator<FetchedMessage> messages) {
 
     int count = 0;
 
+    long oldestProcessed = 0;
     while (messages.hasNext()) {
       FetchedMessage message = messages.next();
       try {
         GenericRecord genericRecord = serializer.toGenericRecord(message.getPayload());
         ILoggingEvent event = serializer.fromGenericRecord(genericRecord);
 
+        if (event.getTimeStamp() < oldestProcessed) {
```

Comment:
we should probably do this after we call ``processor.process(logEvent)`` -- If we get 100 fetched messages and the last one has a old timestamp and if there were exception while processing that event, we will still gauge for that timestamp - which is not accurate as that timestamp event was not processed successfully. might not happen, but better to do there i guess.